### ISSUE TITLE
if/unless option usage unintuitive

### ISSUE THREAD

**Issue by ardation** [CruGlobal/global-registry-bindings#12]
Given the follow code would you expect the recorded to be synced with GR?

```Ruby
class EmailAddress < ActiveRecord::Base
	global_registry_bindings if: :valid_gr_format?

	def valid_gr_format?
		(email =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i).present?
	end
end
```

From this commit https://github.com/CruGlobal/global-registry-bindings/commit/029af37fa523504592e9d0a78ac908216f3f0e38#diff-a8b26bbc553f2c03922a3cc63585e55bR16 it would not sync the record if the email was valid. I think this is not what is expected rather if email was not valid i.e. returned false it should not be run.

This is an issue which we are experiencing in production MPDX here https://github.com/CruGlobal/mpdx_api/commit/18db62b7322cf2fbfa4b4b6f0ef693d3b3b78550#diff-1db36b36de8317130613417fbca17344R45.

**Comment by mattdrees** [comment]
Yeah, this is unintuitive to me as well. `if` and `unless` feel backwards.

**Comment by mattdrees** [comment]
@Omicron7 are these working as you intended? is `if` supposed to mean 'skip-gr-sync-if-this-is-true'?

**Comment by Omicron7** [comment]
I don't remember. Documentation is here
https://github.com/CruGlobal/global-registry-bindings#conditional-push
And specs are here.
https://github.com/CruGlobal/global-registry-bindings/blob/master/spec/options/if_unless_spec.rb

I was certain it was working correctly and I believe its used by a few other projects as well.

**Comment by mattdrees** [comment]
This doc is wrong, according to your specs:
```ruby
class Product < ActiveRecord::Base
  attr_accessor :should_push
  global_registry_bindings if: proc { |model| model.should_push }
end
```
the accessor should be called `:should_not_push`.


This doc is right, but it still feels odd to me to use 'unless' here.
```ruby
class Product < ActiveRecord::Base
  global_registry_bindings unless: :should_push

  def should_push(_model)
    return ::GlobalConfig.gr_enabled?
  end
end
```

**Comment by ardation** [comment]
@Omicron7 I think your intention was for it to be the other way around since you wrote the PR for MPDX where `valid_gr_format?` is meant to return true if the email is valid which should then push it to the GR.

I think there may be other projects suffering from the same issue if that is the case.
